includes/abstracts/abstract-wc-log-handler.php

```diff
@@ -0,0 +1,141 @@
+<?php
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly
+}
+
+if ( ! class_exists( 'WC_Logger' ) ) {
+	include_once( dirname( dirname( __FILE__ ) ) . '/class-wc-logger.php' );
+}
+
+/**
+ * Abstract WC Log Handler Class
+ *
+ * @version        1.0.0
+ * @package        WooCommerce/Abstracts
+ * @category       Abstract Class
+ * @author         WooThemes
+ */
+abstract class WC_Log_Handler {
```

Comment:
To provide the following functionality by default:

1. Threshold and level logic. This reason may be invalidated based on [your other question](https://github.com/woocommerce/woocommerce/pull/12340#pullrequestreview-11564598).
1. Provide default message formatting. A simple coherent message format is something most handlers should be able to leverage. For example, both file and email handlers take advantage of default formatting.
1. Provide default timestamp formatting according to [spec](https://tools.ietf.org/html/rfc5424#section-6.2.3).